### PR TITLE
chore(web): date time field input

### PR DESCRIPTION
**Pull request by mkumbobeaty**
# Overview

## What I've done
- Improved date time field input

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -152,6 +152,7 @@
         "lodash-es": "4.17.21",
         "lru-cache": "8.0.4",
         "mini-svg-data-uri": "1.4.4",
+        "moment-timezone": "^0.5.43",
```
Comment (KaWaite) [inline web/package.json:155]: Please remove the ^ and do yarn again 🙏🏻 If we don't, everyone's local can have different versions and it gets hard to troubleshoot local issues

File: web/src/services/i18n/translations/ja.yml [hunk 2]
```diff
@@ -18,6 +18,11 @@ Edit: 編集
 Color Picker: カラーピッカー
 You have passed the maximum value.: You have passed the maximum value.
 You have passed the minimum value.: You have passed the minimum value.
+Set Time: ''
```
Comment (KaWaite) [inline web/src/services/i18n/translations/ja.yml:21]: 時間設定

File: web/src/services/i18n/translations/ja.yml [hunk 3]
```diff
@@ -18,6 +18,11 @@ Edit: 編集
...
+Date: 作成日時順
+Time: ''
```
Comment (KaWaite) [inline web/src/services/i18n/translations/ja.yml:23]: 時間

File: web/src/services/i18n/translations/ja.yml [hunk 4]
```diff
@@ -18,6 +18,11 @@ Edit: 編集
...
+Time Zone: ''
```
Comment (KaWaite) [inline web/src/services/i18n/translations/ja.yml:24]: 時間帯

File: web/src/services/i18n/translations/en.yml [hunk 5]
```diff
@@ -18,6 +18,11 @@ Edit: Edit
 Color Picker: Color Picker
 You have passed the maximum value.: You have passed the maximum value.
 You have passed the minimum value.: You have passed the minimum value.
+Set Time: ''
+Date: Date
+Time: ''
+Time Zone: ''
+set: ''
```
Comment (KaWaite) [inline web/src/services/i18n/translations/en.yml:25]:
```suggestion
Set Time: Set Time
Date: Date
Time: Time
Time Zone: Time Zone
set: set
```

File: web/src/beta/components/fields/DateTimeField/EditPanel/index.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        <TextWrapper>
+          <Label>{t("Time")}</Label>
+
+          <TextInput className="customTextInput" type="time" value={time} onChange={onTimeChange} />
+        </TextWrapper>
+        <SelectWrapper>
+          <Label>{t("Time Zone")}</Label>
+          <SelectField
+            value={selectedTimezone}
+            className="timezone"
```
Comment (KaWaite) [inline web/src/beta/components/fields/DateTimeField/EditPanel/index.tsx:50]: Is this timezone className doing anything? Since we aren't using any external css files (and you aren't adding any in this PR) that would define a timezone class seems like its maybe a mistake?

File: web/src/beta/components/fields/DateTimeField/EditPanel/hooks.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      const offsetA = parseInt(a.offset);
+      const offsetB = parseInt(b.offset);
+      return offsetA - offsetB;
+    });
+
+    return uniqueTimezones;
+  }, []);
+
+  const zones: string[] = moment.tz.names();
+  const offsetFromUTC: TimezoneInfo[] = getUniqueTimezones(zones);
```
Comment (KaWaite) [inline web/src/beta/components/fields/DateTimeField/EditPanel/hooks.ts:56]: If I am wrong please let me know, but I feel like this will be relatively static, so can you pull these out of the hook? If moment must be used inside a hook or a react component, maybe its fine here. But it seems more like a util that we can import and use once in a useMemo and then it shouldn't re-render at all and keeps the code simpler.

Whether or not you do the above, also please:
- put `const zones: string[] = moment.tz.names();` inside getUniqueTimezones because its not being used anywhere else
- make offsetFromUTC a useMemo

File: web/src/beta/components/fields/DateTimeField/EditPanel/index.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    selectedTimezone,
+    onDateChange,
+    onTimeChange,
+    getUniqueTimezones,
+    onTimezoneSelect,
+    onDateTimeApply,
+  } = useHooks({ onChange });
+
+  const isButtonDisabled = date.trim() === "" || time.trim() === "";
+  const uniqueTimezones = getUniqueTimezones(timezones);
```
Comment (KaWaite) [inline web/src/beta/components/fields/DateTimeField/EditPanel/index.tsx:32]: You are already doing this in the hooks with `offsetFromUTC`. Just pass that from the hooks and use that instead.

File: web/src/beta/components/fields/DateTimeField/EditPanel/index.tsx [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    timezones,
+    selectedTimezone,
+    onDateChange,
+    onTimeChange,
+    getUniqueTimezones,
+    onTimezoneSelect,
+    onDateTimeApply,
+  } = useHooks({ onChange });
+
+  const isButtonDisabled = date.trim() === "" || time.trim() === "";
```
Comment (KaWaite) [inline web/src/beta/components/fields/DateTimeField/EditPanel/index.tsx:31]: Move in to hooks and wrap in useMemo please

File: web/src/beta/components/fields/DateTimeField/EditPanel/index.tsx [hunk 10]
```diff
@@ -0,0 +1,116 @@
+import Button from "@reearth/beta/components/Button";
+import PanelCommon from "@reearth/beta/components/fields/CameraField/PanelCommon";
```
Comment (KaWaite) [inline web/src/beta/components/fields/DateTimeField/EditPanel/index.tsx:2]: Please move this in to fields/common if its not just a CameraField related thing anymore 🙏🏻

File: web/src/beta/components/fields/SelectField/index.tsx [hunk 11]
```diff
@@ -22,6 +22,7 @@ export type Props = {
   // Property field
   name?: string;
   description?: string;
+  className?: string;
```
Comment (KaWaite) [inline web/src/beta/components/fields/SelectField/index.tsx:25]: Since you aren't using the `timezone` className is this change still necessary? Can you clean up any parts related to that that aren't needed? 🙏🏻

Comment (mkumbobeaty) [reply web/src/beta/components/fields/SelectField/index.tsx]: @KaWaite  I come to realize that className is used to override the `PickerWrapper` height and add scroll, I removed this but the select in timezone look ugly, if you may have any suggestion to add the height, overflow and with in this component.

**Review by nina992 — APPROVED** [review]

**Review by KaWaite — APPROVED** [review]
